### PR TITLE
SAK-33720 Try again to join a section of the same category

### PR DESCRIPTION
**Pull request by manuelal**

### CODE REVIEW

File: edu-services/sections-service/sections-impl/sakai/impl/src/java/org/sakaiproject/component/section/sakai/SectionManagerImpl.java [hunk 1]
```diff
@@ -688,7 +688,24 @@ public EnrollmentRecord joinSection(String sectionUuid, int maxSize) throws Role
...
+		List realMembershipInCategory=new ArrayList();
+		// Check membership of groups
+		
+		for(String msic: (List <String>) membershipInCategory)
+		{
+			String realmGroup=msic;
+			Group group = findGroup(realmGroup);
+			Member member = group.getMember(userUid);
+			if(member == null) {
+					if (log.isDebugEnabled()) log.debug("getAuthzUserGroupIds said " + userUid + " is member of " + group.getId() + " but getMember disagrees");
```
Comment (ern) [inline edu-services/sections-service/sections-impl/sakai/impl/src/java/org/sakaiproject/component/section/sakai/SectionManagerImpl.java]: replace with slf4j style log message:
`log.debug("getAuthzUserGroupIds said {} is member of {} but getMember disagrees", userUid, group.getId());`

**Comment by ern** [comment]
@manuelal I am wondering if the original check should be removed/changed so that we can just one check accurately?

**Comment by manuelal** [comment]
@ern  What do you mean by "the original check"?

**Comment by manuelal** [comment]
@ern I have removed the comment "Check membership of groups". Did you mean that?

**Comment by smarquard** [comment]
What is actually going on here to cause this error? Is there some caching somewhere that is causing the two checks to differ, i.e.

    List membershipInCategory = authzGroupService.getAuthzUserGroupIds(categorySections, userUid);

vs

````
Group group = findGroup(realmGroup);
Member member = group.getMember(userUid);
````

**Comment by juanjmerono** [comment]
From the documentation:

> List<AuthzGroup>getAuthzUserGroupIds(ArrayList<String> authzGroupIds, String userid)
> Access a list of AuthzGroups which contain a specified userid NOTE: This call is backed by a cache.

**Comment by juanjmerono** [comment]
@ern are you agree to merge this? I think now is ready.

**Pull request merged**